src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.CodeAction_Method.cs

```diff
@@ -14,6 +15,63 @@ namespace Microsoft.CodeAnalysis.ImplementInterface
 {
     internal abstract partial class AbstractImplementInterfaceService
     {
+        private sealed class InterfaceReplacerWithContainingTypeVisistor : SymbolVisitor<ISymbol>
+        {
+            private readonly INamedTypeSymbol _intrerfaceSymbol;
+            private readonly INamedTypeSymbol _containingTypeSymbol;
+
+            public InterfaceReplacerWithContainingTypeVisistor(INamedTypeSymbol interfaceSymbol, INamedTypeSymbol containingTypeSymbol)
+            {
+                Debug.Assert(interfaceSymbol.TypeKind == TypeKind.Interface);
+                Debug.Assert(containingTypeSymbol.TypeKind is TypeKind.Class or TypeKind.Struct);
+                _intrerfaceSymbol = interfaceSymbol;
+                _containingTypeSymbol = containingTypeSymbol;
+            }
+
+            public override ISymbol VisitMethod(IMethodSymbol symbol)
+            {
+                Debug.Assert(symbol.MethodKind == MethodKind.UserDefinedOperator);
+                if (symbol.Parameters.Length == 0)
+                {
+                    return symbol;
+                }
+
+                var updatedParameters = ImmutableArray.CreateBuilder<IParameterSymbol>(initialCapacity: symbol.Parameters.Length);
+                foreach (var parameter in symbol.Parameters)
+                {
+                    updatedParameters.Add((IParameterSymbol)VisitParameter(parameter));
+                }
+
+                var returnType = symbol.ReturnType;
+                if (returnType is INamedTypeSymbol namedTypeSymbol)
+                {
+                    returnType = (INamedTypeSymbol)VisitNamedType(namedTypeSymbol);
+                }
+
+                return CodeGenerationSymbolFactory.CreateMethodSymbol(symbol, parameters: updatedParameters.ToImmutable(), returnType: returnType);
+            }
+
+            public override ISymbol VisitParameter(IParameterSymbol symbol)
+            {
+                if (symbol.Type is INamedTypeSymbol namedTypeSymbol)
+                {
+                    return CodeGenerationSymbolFactory.CreateParameterSymbol(symbol, type: (INamedTypeSymbol)VisitNamedType(namedTypeSymbol));
+                }
+
+                return symbol;
+            }
+
+            public override ISymbol VisitNamedType(INamedTypeSymbol symbol)
+            {
+                if (symbol.Equals(_intrerfaceSymbol, SymbolEqualityComparer.Default))
+                {
+                    return _containingTypeSymbol;
+                }
+
+                return symbol;
+            }
+        }
```

Comment:
1.this is all interesting.  is this the lgoic the compiler uses here?
2. can we have tests with generic classes/interfaces?

Reply:
1. Probably not :smile: I believe things are more complicated than what I've here, specifically for generics (I'm actually not able to implement the non-generic version :smile: see https://github.com/dotnet/roslyn/issues/53927#issuecomment-856699046).
2. Sure.

Reply:
the symbolvisitor seems overkill since we're not recurisng past return-type/parameter.  seems like we could just have a simple method that accomplishes the same.